### PR TITLE
Doco how to update xonsh

### PR DESCRIPTION
**Pull request by halloleo**
Doco about how to update xonsh and how to unset environment variables

Relates to later comments in issue #2928.

### CODE REVIEW

**Comment by halloleo** [comment]
Ups, something went wrong here! Somehow the changes of the old branch/PR `doco-arbitrary-paths-in-vox` are unwillingly included in this branch `doco-how-to-update-xonsh`. Not sure how to untangle them...

File: docs/customization.rst [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+Updating xonsh
+===========
+
+You can update xonsh from within itself with the following command:
+
+.. code-block:: console
+
+   $ xpip install --update xonsh
+
+``xpip`` is a predefined alias pointing to the pip command assosiated with the python executable running this xonsh.
```
Comment (gforsyth) [inline docs/customization.rst]:
```suggestion
``xpip`` is a predefined alias pointing to the pip command associated with the python executable running this xonsh.
```

**Comment by gforsyth** [comment]
Hey @halloleo -- I think we've all been down some kind of git hell-hole at one point or another.  

I would suggest, generally, that if you want to clean up commits AND there is important discussion in a pull request, then rebasing/squashing makes sense -- then you keep the discussion and the evolution but not the "dirty" history (this is also a good argument for enabling squash on merge with github).

In the case of a relatively small docs update, it might not be worth the trouble of doing all of that and you can create a new PR.  

It's up to you -- if you feel like there's a story here that might be helpful to others, then I would fix the current branch.  If not, then go ahead and create a new one.

Just my two cents.

**Comment by scopatz** [comment]
Any updates on this @halloleo - and thanks for putting it in!

**Comment by halloleo** [comment]
I deleted this PR (because of some git stuff-up) and re-issued it as #2964.

**Pull request closed without merging**